### DISCUSSION TITLE
Null default field for select form component

### DISCUSSION THREAD

**Discussion by julesbouton** [sqlpage/SQLPage#539] [Q&A]
Hi,
I come up (again) with a question. There is a nice behaviour in a multiple searchable selector: As a default, the field is left empty and if submitted as it is, it returns a NULL value. E.g.:

```sql
SELECT 'Suffix' AS name,
        'Type of suffix' as label,
        'select' as type,
        TRUE     as multiple,
        TRUE     as searchable
...
```

On the opposite, a multiple selector or a simple selector automatically submits its first value, and it is hard to obtain a default NULL value. (for instance if the user isn't required to select something)

```sql
SELECT 'Suffix' AS name,
        'Type of suffix' as label,
        'select' as type,
...
```

I think that it would be nice to have the first behaviour also for simple selectors (maybe with an additional row-level parameter?)

**Comment by lovasoa** [comment]
If you want to allow your users to leave the field empty, you can add an empty option at the beginning of the options of your simple select field.

**Reply by julesbouton** [reply]
That's what I do when I supply a JSON string containing the options: `{'"value":""'}` works well. But it's less evident if the options come from a table for instance.

**Reply by lovasoa** [reply]
You can always do something like this:

```sql
select 'select' as type,
  (
    select json_group_array(json_object('label', name, 'value', id))
    from (
      select 'Choose a value' as name, '' as id
      union all
      select name, id from demo
    ) as t
  ) as options
```
